### ISSUE TITLE
Too many HttpClient (TCP connections)

### ISSUE THREAD

**Issue by jesuissur** [BingAds/BingAds-dotNet-SDK#94]
Hi there,

That seems a bad practice to create a new HttpClient every time as stated in [https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netframework-4.7.2#remarks](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netframework-4.7.2#remarks)

https://github.com/BingAds/BingAds-dotNet-SDK/blob/96449d99efc36ae9d769733847ebf88948ce6d8e/BingAdsApiSDK/Internal/HttpService.cs#L65

Thanks,
Phil

**Comment by qitia** [comment]
thanks @jesuissur we push new version 13.0.9 today, try it out?

**Comment by jesuissur** [comment]
Hi,

We've just tried the new version and it does not work in our scenario.  Our setup is the following:
1. We got an Azure Function .Net core 3.1
2. It uses assemblies built in .Net framework 4.8.  And those assemblies use the BingAds SDK.

So in this scenario, the SDK use the old HttpClient from .Net 4.8 ([without the new and efficient SocketsHttpHandler](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler?view=netcore-3.1#remarks)).  We run a few benchmark tests and the http connections still go through the roof under load.

We [forked the project](https://github.com/DashThis/BingAds-dotNet-SDK) and introduced a new way to provide our own HttpClient to the SDK.  In our case, we then can provide a HttpClient (coming from .Net core 3.1) created through the HttpClientFactory with the SocketsHttpHandler.  It looks like that from our side in our app startup:
`Microsoft.BingAds.HttpClientCustomFactory.ReplaceHttpClientFactory(() => GetHttpClientWithEfficientSocketsUsage());`

Then our benchmark were great with the connections count staying under the defined [MaxConnectionsPerServer](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.maxconnectionsperserver?view=netcore-3.1#System_Net_Http_SocketsHttpHandler_MaxConnectionsPerServer).

Our solution may not be exactly the way you'd like, but we think having a way to provide an external custom "HttpClientFactory" would be a great improvement overall and the way to go for us.

Let me know if we can help anyhow.
Thank you,
Phil
